plugins/e57/io/Scan.cpp

```diff
@@ -141,13 +141,16 @@ void Scan::decodeHeader()
     //        (double value) i.e 65535/(65535-0)= 1.
 
     std::fill_n(m_rescaleFactors, pdal::Dimension::COUNT, 1.0); //Re-initialize all rescale factors to 1.0
-    for (auto& field : supportedFields)
+    auto rescaleableFields = pdal::e57plugin::rescalableE57Types();
+    for (auto& field : rescaleableFields)
     {
         auto minmax = std::make_pair(0.0, 0.0);
         if (pdal::e57plugin::getLimits(*m_rawData, field, minmax))
+        {
             m_rescaleFactors[(int)pdal::e57plugin::e57ToPdal(field)] =
                 (std::numeric_limits<uint16_t>::max)() /
```

Comment:
This assumes that the target of the rescaling is 65534 which will not always be the case. Maybe you can change `rescalableE57Types` into a map and store the target scale there? Alternatively you can ask PDAL for the type of that dimension and use that instead of uint16_t